launch/pipeline/service.py

```diff
@@ -45,10 +65,26 @@ def _init_callable_service(self):
             ), "`kwargs` is given, but the service is not a class"
             self._callable_service = self.service
 
+    def set_make_request(
+        self, make_request: Callable, force: bool = False
+    ) -> None:
+        if force:
+            self.make_request = make_request
+
     def call(self, req: Any) -> Any:
-        self._init_callable_service()
-        assert self._callable_service, "`_callable_service` is not initialized"
-        return self._callable_service(req)
+        if self.make_request:
+            return self.make_request(
+                servable_id=self.service_name,
+                local_fn=self.service,
+                args=[req],
+                kwargs={},
+            )
+        else:
+            self._init_callable_service()
```

Comment:
This seems a bit complicated and a little out-of-order: wouldn't we want to initialize the service first so we can put it into a state where it can always execute its `call`? Why not do this initialization at the moment a `SingleServiceDescription` instance is created? Are there any benefits to having a `SingleServiceDescription` exist in this intermediate, not-ready-to-execute state?

Reply:
If this `ServiceDescription` class eventually gets serialized, maybe we initialize lazily because of https://scale.atlassian.net/wiki/spaces/EPD/pages/31195169/Celery+Guide#Serving-a-torch-module-on-a-GPU-worker (i.e. in celery we need to initialize pretty late)? That's my guess anyways, but I'm not super sure.

Reply:
Yeah we were discussing this distinction between client-side vs. server-side initialization at some point. I guess right now is the forcing function for us to reach parity between these two scenarios. If we want initialization in the constructor for both local and remote, then we'll want to serialize and transmit the class definition instead of an object.